portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,92 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+let root = document.documentElement;
+
+const secretText = document.querySelector("#secret-text"); 
+const mePic = document.querySelector("#me-pic"); 
+const tomiePic = document.querySelector("#tomie-pic"); 
+const mainHeading = document.querySelector("#main-heading"); 
+
+//fade in main text
+const mainText = document.querySelector(".main-text");
+window.addEventListener('load', e=> {
+    mainText.classList.add("fade-text-in"); 
+})
+
 /**
- * Adds a random greeting to the page.
+ * Keeps track of mouse and updates our special, circle pointer to
+ * follow cursor movement. 
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+const mouseCursor = document.querySelector(".cursor");
+window.addEventListener('mousemove', e=> {
+    mouseCursor.style.top = e.pageY + "px"; 
+    mouseCursor.style.left = e.pageX + "px"; 
+})
+
+/**
+ * Animates the mouse as it interacts with section headers. 
+ */
+secretMessages = ["...", "....", "Would you like to see something fun?"]; 
+var messageIndex = 0; 
+
+const sections = document.querySelectorAll(".sections li");
+sections.forEach(section => {
+    section.addEventListener("mouseover", e=> {
+        mouseCursor.classList.add("grow-cursor"); 
+        section.classList.add("color-text"); 
+    }); 
+    section.addEventListener("mouseleave", e=> {
+        mouseCursor.classList.remove("grow-cursor");
+        section.classList.remove("color-text"); 
+    }); 
+    section.addEventListener("click", e=> {
+        console.log(messageIndex); 
```

Comment:
Don't submit log statements unless they have value in production.  For instance, logging errors is OK.

Also in other places.